core/src/collections.rs

```diff
@@ -26,29 +28,107 @@ impl MongoCollections {
     fn next(&mut self) -> Result<bool> {
-        unimplemented!()
+        if self.current_collection_list.is_none() {
+            if self.collections_for_db_list.as_ref().unwrap().is_empty() {
+                return Ok(false);
+            }
+            self.current_collection_list =
+                Some(self.collections_for_db_list.as_mut().unwrap().remove(0));
+        }
+        loop {
+            if self
+                .current_collection_list
+                .as_mut()
+                .unwrap()
+                .collection_list
+                .advance()?
+            {
+                // Cursor advance succeeded, update current CollectionSpecification
+                self.current_collection = Some(
+                    self.current_collection_list
+                        .as_ref()
+                        .unwrap()
+                        .collection_list
+                        .deserialize_current()
+                        .unwrap(),
+                );
+                return Ok(true);
+            }
+            if self.collections_for_db_list.as_ref().unwrap().is_empty() {
+                return Ok(false);
+            }
+            // Current cursor exhausted, try next database in list
+            self.current_collection_list =
+                Some(self.collections_for_db_list.as_mut().unwrap().remove(0));
+        }
     }
 
     // Get the BSON value for the given colIndex on the current CollectionSpecification.
     // Fails if the first row as not been retrieved (next must be called at least once before getValue).
-    fn get_value(&self, _col_index: u16) -> Result<Option<&Bson>> {
+    fn get_value(&self, col_index: u16) -> Result<Option<Bson>> {
         // The mapping for col_index <-> Value will be hard-coded and handled in this function
-        // 1-> current_coll_list.0
-        // 2 -> current_coll.name
-        // 3 -> current_coll.CollectionType
-        unimplemented!()
+        // 1-> current_collection_list.database_name
+        // 2-> Schema name; NULL as it is not applicable
+        // 3 -> current_collection.name
+        // 4 -> current_collection.collection_type
+        match col_index {
+            1 => Ok(Some(Bson::String(
+                self.current_collection_list
+                    .as_ref()
+                    .unwrap()
+                    .database_name
+                    .clone(),
+            ))),
+            2 => Ok(Some(Bson::Null)),
+            3 => Ok(Some(Bson::String(
+                self.current_collection.as_ref().unwrap().name.clone(),
+            ))),
+            4 => Ok(Some(Bson::String(format!(
+                "{:?}",
+                self.current_collection.as_ref().unwrap().collection_type
+            )))),
+            _ => Ok(Some(Bson::Null)),
+        }
     }
 }
+
+// Replaces SQL wildcard characters with associated regex
+// Returns a doc applying filter to name
+fn to_name_regex(filter: &str) -> Document {
```

Comment:
This can be improved to be in line with the changes made in the JDBC driver SQL-567.  I'll create a ticket to do that.